### PR TITLE
Lets antlr

### PR DESCRIPTION
**Pull request by stephen-hawley**
Added code to get antlr and build the grammar.
CS code generation support will follow

### CODE REVIEW

File: swiftinterfaceparser/Makefile [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+JDK_VERSION=$(shell ls /usr/local/Cellar/openjdk | sort -V | tail -n 1)
+JDK_PATH="/usr/local/Cellar/openjdk/$(JDK_VERSION)"
+JAVA_HOME=$(JDK_PATH)
+
+all: $(ANTLRJAR) grammar
+
+
+# download antlr on demand
+$(ANTLRJAR):
+	curl --create-dirs --output $(ANTLRJAR) https://www.antlr.org/download/antlr-4.9.1-complete.jar
```
Comment (mandel-macaque) [inline swiftinterfaceparser/Makefile:20]: We might want to follow redirects here using the '-L' option.

File: swiftinterfaceparser/Makefile [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+CLASSPATH=.:$(ANTLRJAR)
+
+GRAMMAR=SwiftInterface.g4
+
+CSOUTDIR=CSParser
+JAVAOUTDIR=JavaParser
+
+JDK_VERSION=$(shell ls /usr/local/Cellar/openjdk | sort -V | tail -n 1)
+JDK_PATH="/usr/local/Cellar/openjdk/$(JDK_VERSION)"
+JAVA_HOME=$(JDK_PATH)
```
Comment (rolfbjarne) [inline swiftinterfaceparser/Makefile:13]: These variables don't seem to be used.

**Review by rolfbjarne — APPROVED** [review]

File: swiftinterfaceparser/Makefile (hunk 2 above)
Comment (stephen-hawley) [reply swiftinterfaceparser/Makefile:13]: `JAVA_HOME` is used by `java` and `java`

File: swiftinterfaceparser/Makefile [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+$(ANTLRJAR):
+	curl --create-dirs --output $(ANTLRJAR) https://www.antlr.org/download/antlr-4.9.1-complete.jar
+
+# this just builds the grammar to verify that it's correct
+# The output java files can be user with antlr's built-in tools for
+# printing parse trees from test files
+
+grammar: $(GRAMMAR) $(JAVAOUTDIR)
+	$(ANTLR) -o $(JAVAOUTDIR) $(GRAMMAR)
+	javac -classpath $(CLASSPATH) $(JAVAOUTDIR)/*.java
```
Comment (mandel-macaque) [inline swiftinterfaceparser/Makefile:28]: Do you want to see all the noise? I suppose you do :)

**Review by mandel-macaque — APPROVED** [review]
Maybe add the follow redirect flag

**Pull request merged**